[master] Porting #50416 to master

garethgreenaway opened the pull request:
Porting #50416 to master

Ch3LL reviewed (changes requested):
needs testcase. i'll look into writing one

waynew committed:
Add tests for nilrt_ip._load_config

Currently there's an issue where providing non-text default values will
cause bad news. These tests exhibit the bad behavior (and some of the
existing good behavior)

waynew committed:
Fix issue in nilrt_ip

Before, passing an int or anything that could not be sliced and indexed
as the default value would cause issues when there are no matching
config sections. Now, we will only unquote values when the option is
present in the config file.

waynew committed:
Remove python2-isms

waynew commented:
@Ch3LL I went ahead and added tests during today's test clinic - also improved the code a bit (and, of course, the dropping py2 code 🙃 )

waynew committed:
Add changelog entry
